### DISCUSSION TITLE
Unable to connect via public IP address

### DISCUSSION THREAD

**Discussion by coyohti** [evennia/evennia#2467] [Evennia Questions] [answered]
Hello!

EDIT: This refers to access via the web client.

I am trying to get my Evennia server set up to be accessible outside my home network. Unfortunately, I have not had much luck and my ability to troubleshoot this further on my own has reached its limit.

Currently I have Evennia running on a Raspberry Pi 3 Model 3 using Buster (latest OS).

I have tried everything I could mentioned on the [Online Setup](https://www.evennia.com/docs/latest/Online-Setup.html) page.  Originally, I couldn't even connect to Evennia via my home network but setting up port forwarding on the router seems to have helped that, at least.

As a test I installed Apache and it worked out of the box. I am completely unfamiliar with Twisted and how it differs so now I'm at a complete loss.

If anyone has any further suggestions I am most grateful! Thank you!

EDIT: I also installed everything on a Windows machine to test it out and have the same result - unable to connect via the outside.

**Comment by Griatch** [comment, marked as answer]
Apache will not really help you if you aim to use the Evennia webserver + webclient. Evennia is its own webserver, so if you don't want Apache for something in particular, I suggest turning it off and letting Evennia itself handle the web connectivity on your raspberry.

If you can connect to Evennia locally, or even better from your local network (including website + webclient with websockets), then all is fine on the Evennia end. Your issue is in routing from the ports of your server to the outside world.

The external ports of your router needs to be set up to forward the incoming port to your raspberry and the ports Evennia uses. You can start by routing port 4000 on your router to 4000 on the raspberry, then see if you can connect a telnet client to `<your_external_ip>, port 4000`, then continue with 4001 (the web port) and 4002 (websockets) (go to `http://<your_external_ip>:4001` - you should see the game website). You may also need to open those ports in your firewall on the router (probably not on your raspberry if you can already connect to it from your local network).

If you have no other webserver you can in the future route Evennia's 4001 to the external port 80 (default HTTP) - if so you don't need to give the port explicitly. But start with the explicit ports first. Also make sure you know what the outward-facing IP-address of your router actually is - your ISP may change this on you from time to time (it's also never `192.168.x.x`, that's a local address).  

Alas, exactly how to do this varies from router to router. It doesn't sound like this is an Evennia issue per se though.

**Reply by coyohti** [reply]
Thank you so much for your reply!

I should have clarified - testing with Apache was just the "easiest" way I could think of to check if I could access the Pi from the outside at all. Once I saw it was successful I removed Apache.

After reading this I'm sure my issue was with the port numbers. I was confused by the note on the Online Setup page which starts with "...the router’s Internet-facing ports do not have to have to have the same numbers as your computer’s (and Evennia’s) ports". Somehow I interpreted this as meaning Evennia would sort it out on its own. Oops! In hindsight I should have tried opening up the 400* ports mentioned in the docs on my firewall as part of my troubleshooting.

That said, I had decided to start with a fresh install of the OS and everything and am now running into other issues whereas the first install had gone so swimmingly. In this case it is definitely an RPi and/or user error issue.

Again, thanks for the help~!
